aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -3456,13 +3481,19 @@ static void linalg_lstsq_out_info(
       at::sum_out(residuals, raw_residuals, /*dim=*/-2, /*keepdim=*/false, /*dtype*/real_dtype);
     }
   }
-  auto solution_view = solution.narrow(/*dim=*/-2, /*start=*/0, /*length*/n);
-  // manually restride original
-  solution.set_(solution.storage(), solution_view.storage_offset(), solution_view.sizes(), solution_view.strides());
+  }
+  if (solution.size(-2) >= n) {
+    auto solution_view = solution.narrow(/*dim=*/-2, /*start=*/0, /*length*/n);
```

Comment:
@lezcano 

I added this, the tests for `test_linalg_lstsq` was failing for :
` python test_linalg.py -k test_linalg_lstsq_cuda_float32`

with tensors like:

```
torch.Size([2, 1]) is a shape
a contents: tensor([[0.1540],
        [0.9887]], device='cuda:0')
```